### ISSUE TITLE
Scheduler - Context Menu

### ISSUE THREAD

**Issue by dxbykov** [DevExpress/DevExtreme#3373]
# The Problem 

It is quite common to use the context menu in modern web scheduling applications. For instance, a user can create, update and delete appointments via the context menu. The DevExtreme [Scheduler](https://js.devexpress.com/Demos/WidgetsGallery/Demo/Scheduler/Overview/jQuery/Light/) does not have a convenient API for such use-cases until v18.1.

# The Solution 

In v18.1, we introduced the new `onAppointmentContextMenu` and `onCellContextMenu` events that allow you to create and configure a context menu. The context of the click is passed as arguments to the corresponding event handlers.

```js
        onAppointmentContextMenu: function(e) {
            var appointmentData = e.appointmentData;
                  targetedAppointmentData = e.targetedAppointmentData;

            ...

        },
        onCellContextMenu: function(e) {
            var scheduler = e.component,
                  cellData = e.cellData;

	    ...
        },

```

The event arguments are similar to the [onAppointmentClick](https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#onAppointmentClick ) and [onCellClick](https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#onCellClick) event arguments.

# Try It And Give Us Feedback 

## Live Sandbox

[jQuery](https://codepen.io/dxbykov/pen/eMvpoe)
[Angular](https://embed.plnkr.co/yVzblB5UaY1n8PG9jjfH/)

![scheduler-context-menu](https://user-images.githubusercontent.com/8307562/37672001-06202688-2c7e-11e8-85a1-fd2f51d0d54b.png)

## Installation

Follow the installation guide from the [release page](https://github.com/DevExpress/DevExtreme/releases) and leave us feedback on this topic.

**Comment by kimrohdin** [comment]
Great iniative! We have implemented our own contex-menu for now. An really important feature for us, is that you can select multiple cells, right click and then get both start and end time for the selection. Will this be supported?

**Comment by dxbykov** [comment]
Thanks for the feedback. Yes, we expect to implement this in 18.1.

**Comment by dxbykov** [comment]
Thanks to everybody who participated in this discussion or gave us early feedback by other means. The feature is now available in the [v18.1](https://github.com/DevExpress/DevExtreme/releases/tag/18.1.3) release. I'm closing this thread. In the case of bugs or questions, feel free to create new GitHub [issues](https://github.com/DevExpress/DevExtreme/issues) or tickets in our [Support Center](https://www.devexpress.com/Support/Center/).
